include/swift/AST/DiagnosticsParse.def

```diff
@@ -131,17 +131,17 @@ ERROR(lex_unicode_escape_braces,none,
 ERROR(lex_illegal_multiline_string_start,none,
       "multi-line string literal content must begin on a new line", ())
 ERROR(lex_illegal_multiline_string_end,none,
-      "multi-line string literal content must not be on the delimiter's line", ())
-ERROR(lex_insufficient_string_indent,none,
-      "string literal content is insufficiently indented", ())
-ERROR(lex_unexpected_tab_in_string_indent,none,
-      "string literal content is indented with a tab where a space is expected", ())
-ERROR(lex_unexpected_space_in_string_indent,none,
-      "string literal content is indented with a space where a tab is expected", ())
+      "multi-line string literal closing delimiter must begin on a new line", ())
+ERROR(lex_inconsistent_string_indent,none,
+      "line does not start with same indentation as closing delimiter of multi-line string literal", ())
+NOTE(note_insufficient_string_indent,none,
+      "unexpectedly found non-indentation character here", ())
+NOTE(note_unexpected_char_in_string_indent,none,
+      "unexpected %select{space|tab}0 in indentation here", (bool))
+NOTE(note_matching_indentation_char_here,none,
+      "should match %select{space|tab}0 in same position before closing delimiter", (bool))
```

Comment:
The problem with one error + multiple notes is that, when you're using Xcode, the notes don't show up in the editor. So the Xcode experience would be that the entire literal is flagged as having indentation errors and there's a magic fix-it that tries to correct all of them, but no easy way to see which lines are erroneous.

Maybe we should split the difference: Merge together errors in adjacent lines, considering the common whitespace prefix[1] to be the indentation to remove. So if you wrote this (where the delimiter is indented with a tab, but the other lines have spaces):

```swift
let xml = """
    <html>
        <p>Hello, world!</p>
    </html>
	"""
```

You'd get something like:

```
example.swift:2:1: error: space in next 3 lines of multi-line string literal does not match tab indenting closing delimiter
    <html>
^
example.swift:3:1: note: also here
        <p>Hello, world!</p>
^
example.swift:4:1: note: also here
    </html>
^
example.swift:5:1: note: should match character here
        """
^^^^^^^^
example.swift:2:1 note: change indentation in these lines
    <html>
^~~~
        
        <p>Hello, world!</p>
^~~~
         
    </html>
^~~~~
        
```

([1] "Common whitespace prefix" is perhaps too simple an algorithm, since it would not detect when parts of a string were pasted from two different sources which were incompatible with the ending delimiter in different ways, but we could always switch to something that's more clever later on.)